src/plugins/analysis/qemu_exec/code/qemu_exec.py

```diff
@@ -216,34 +224,124 @@ def process_qemu_job(file_path: str, arch_suffix: str, root_path: Path, results_
+            with suppress(APIError):
+                container.stop()
+            container.remove()
+
+
+def process_docker_output(docker_output: Dict[str, Dict[str, str]]) -> Dict[str, Dict[str, str]]:
+    process_strace_output(docker_output)
+    replace_empty_strings(docker_output)
+    merge_similar_entries(docker_output)
+    return docker_output
+
+
+def decode_output_values(result_dict: Dict[str, Dict[str, Union[str, int]]]) -> Dict[str, Dict[str, str]]:
+    result = {}
+    for parameter in result_dict:
+        for key, value in result_dict[parameter].items():
+            if isinstance(value, str) and key != 'error':
+                try:
+                    str_value = b64decode(value.encode()).decode(errors='replace')
+                except binascii.Error:
+                    logging.warning('Error while decoding b64: {}'.format(value))
+                    str_value = 'decoding error: {}'.format(value)
+            else:
+                str_value = str(value)
+            result.setdefault(parameter, {})[key] = str_value
     return result
 
 
-def get_docker_output(arch_suffix: str, file_path: str, root_path: Path) -> Optional[str]:
-    call = 'docker run --rm --net=none -v {}:/opt/firmware_root {} {} {}'.format(
-        root_path, DOCKER_IMAGE, arch_suffix, file_path)
-    response, return_code = execute_shell_command_get_return_code(call, timeout=TIMEOUT)
-    if return_code != 0:
-        if 'timed out' in response:
-            logging.warning('encountered timeout while trying to run docker container')
-        else:
-            logging.warning('encountered process error while trying to run docker container')
-        return None
-    return response
+def _strace_output_exists(docker_output):
+    return (
+        'strace' in docker_output
+        and 'stdout' in docker_output['strace']
+        and bool(docker_output['strace']['stdout'])
+    )
 
 
-def parse_docker_output(docker_output: str) -> dict:
-    result = parse_docker_output_options(docker_output)
-    merge_similar_entries(result)
-    result.update(parse_docker_output_strace(docker_output))
-    return result
+def process_strace_output(docker_output: dict):
+    docker_output['strace'] = (
+        zlib.compress(docker_output['strace']['stdout'].encode())
+        if _strace_output_exists(docker_output) else {}
+    )
+
+
+def result_contains_qemu_errors(docker_output: Dict[str, Dict[str, str]]) -> bool:
+    return any(
+        contains_docker_error(value)
+        for parameter in docker_output
+        for value in docker_output[parameter].values()
+    )
+
+
+def contains_docker_error(docker_output: str) -> bool:
+    return any(error in docker_output for error in QEMU_ERRORS)
+
+
+def replace_empty_strings(docker_output: Dict[str, object]):
+    for key in list(docker_output):
+        if key == ' ':
+            docker_output[EMPTY] = docker_output.pop(key)
 
 
 def merge_similar_entries(results_dict: Dict[str, Dict[str, str]]):
```

Comment:
Ugh. I did not review this initially, right?
I would recommend a better name.
One, similar does not fit when you check for equality.
Also maybe rename some variables to make what this does more explicit.